modules/040-node-manager/hooks/handle_node_templates.go

```diff
@@ -186,6 +186,15 @@ func nodeTemplatesHandler(input *go_hook.HookInput) error {
 				}
 			}
 
+			// Migration (remove in d8 1.38): adding 'control-plane' node role to prepare for k8s 1.25
+			const (
+				controlPlaneLabelKey = "node-role.kubernetes.io/control-plane"
+				masterLabelKey       = "node-role.kubernetes.io/master"
+			)
+			if _, ok := nodeObj.Labels[masterLabelKey]; ok {
+				nodeObj.Labels[controlPlaneLabelKey] = ""
```

Comment:
Why we do this migration in 040-node-manager ? Why not to do this in global-hooks ?

Reply:
Agreed to create global hook and this code as an additional safety measure